### PR TITLE
Preparation for variable rate heartbeats

### PR DESCRIPTION
**Pull request by espadolini**
This PR replaces the `MultiInterval` used by the inventory controller with a lightweight wrapper around a `time.Timer` that doesn't make use of goroutines, gets rid of "control log" inventory pings (as the control log mechanism is not sound given the changes to instance heartbeats, and since nothing uses it, there's no reason to exercise it) so we no longer need a method to trigger instance heartbeats from the outside of the inventory, removes an unstable envvar that disables heartbeat keepalives (the largest setup we know of didn't end up needing it) and prepares the inventory controller to make use of variable rate heartbeats, which are going to be per-protocol in a later PR (that will likely not get backported, unlike this one).

### CODE REVIEW

**Review by zmb3 — APPROVED** [review]

File: lib/inventory/controller.go [hunk 1]
```diff
@@ -426,15 +418,35 @@ func (c *Controller) handleControlStream(handle *upstreamHandle) {
...
+					})
+				}
+				if kubeKeepAliveTicker == nil && m.KubernetesServer != nil {
+					kubeKeepAliveTicker = jitteredticker.New(jitteredticker.Params{
+						FirstInterval: retryutils.HalfJitter(c.serverKeepAlive),
+						FixedInterval: c.serverKeepAlive,
 						Jitter:        retryutils.SeventhJitter,
 					})
-					keepAliveNeedInit = false
 				}
```
[the remark this answers is not in the document]
Comment (espadolini) [reply lib/inventory/controller.go:450]: All conditions have to be checked, nothing prevents the `proto.InventoryHeartbeat` from holding more than one heartbeat, and we check and apply all of them individually in `handleHeartbeatMsg`.

I don't believe there's a meaningful difference between checking a local variable (which might be stack allocated) and going through `m` - but if there were, checking the local variable is likely faster.

File: lib/inventory/internal/jitteredticker/jitteredticker.go [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+	Jitter retryutils.Jitter
+
+	clock clockwork.Clock
+}
+
+// New returns a new, running [*JitteredTicker].
+func New(p Params) *JitteredTicker {
+	if p.clock == nil {
+		p.clock = clockwork.NewRealClock()
+	}
```
[the remark this answers is not in the document]
Comment (espadolini) [reply lib/inventory/internal/jitteredticker/jitteredticker.go:50]: Nothing breaks or panics if the values are left to 0 or are set negative - which is more than what `time.NewTicker` does, for example.

File: lib/inventory/controller.go [hunk 3]
```diff
@@ -345,6 +317,33 @@ func (c *Controller) testEvent(event testEvent) {
...
+	var appKeepAliveTicker *jitteredticker.JitteredTicker
+	var dbKeepAliveTicker *jitteredticker.JitteredTicker
+	var kubeKeepAliveTicker *jitteredticker.JitteredTicker
+	defer func() {
+		// this is a function expression because the variables are initialized
+		// later and we want to call Stop on the initialized value (if any)
+		sshKeepAliveTicker.Stop()
+		appKeepAliveTicker.Stop()
+		dbKeepAliveTicker.Stop()
+		kubeKeepAliveTicker.Stop()
```
[the remark this answers is not in the document]
Comment (espadolini) [reply lib/inventory/controller.go:344]: Stop and Next deliberately check for a nil receiver to enable this exact pattern.

**Review by kopiczko — APPROVED** [review]

File: lib/inventory/internal/jitteredticker/jitteredticker.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (espadolini) [reply lib/inventory/internal/jitteredticker/jitteredticker.go:50]: Well, it's documented behavior 😅

The plan (for a follow up PR) is to pass along the current fixed interval as well as an optional VariableDuration for each protocol, and letting the variable one override the fixed one lets us not have to care about setting the fixed one to 0 or something like that.

**Review by fspmarshall — APPROVED** [review]
LGTM overall.  The biggest thing that worries me is accidental misuse of the `JitteredTicker`.  I think it'll be too easy for folks to assume it works like a `time.Ticker` or the existing `Interval` helper, and not be sufficiently diligent about resetting.

What do you think about naming it something like `Delay` instead?  And maybe rename next/advance to something that more clearly implies that this type is more like a wrapper around timer than it is a wrapper around a ticker... Ex:

```golang
myDelay := delay.New(...)

now := <-myDelay.Elapsed()

myDelay.ResetFrom(now)
```

**Comment by espadolini** [comment]
I renamed the type to `Delay` and the channel method to `Elapsed` rather than `Next`, but I left `Advance` because it fits and because `ResetFrom` might give the idea that it's callable in more situations than it should.

> I think it'll be too easy for folks to assume it works like a time.Ticker or the existing Interval helper, and not be sufficiently diligent about resetting.

TBH I'm not entirely sure how you can misuse a Delay without it immediately deadlocking (or very obviously overticking, in the best case scenario), but the subtlety needed in its use is the reason why it's an internal package in `lib/inventory` rather than just being in `utils` or something like that - the inconvenience of hoisting the logic up in the user's select loop is justified if there's a tangible benefit in having fewer running goroutines, which is not true often.

**Pull request merged**